plugin/src/App/Components/Studio/StudioPluginGui.lua

```diff
@@ -0,0 +1,68 @@
+local Rojo = script:FindFirstAncestor("Rojo")
+
+local Roact = require(Rojo.Roact)
+
+local StudioPluginContext = require(script.Parent.StudioPluginContext)
+
+local e = Roact.createElement
+
+local StudioPluginGui = Roact.PureComponent:extend("StudioPluginGui")
+
+StudioPluginGui.defaultProps = {
+	initDockState = Enum.InitialDockState.Right,
+	active = false,
+	overridePreviousState = false,
+	floatingSize = Vector2.new(0, 0),
+	minimumSize = Vector2.new(0, 0),
+	zIndexBehavior = Enum.ZIndexBehavior.Sibling,
+}
+
+function StudioPluginGui:render()
+	return e(StudioPluginContext.Consumer, {
+		render = function(plugin)
+			if not self.pluginGui then
+				local floatingSize = self.props.floatingSize
+				local minimumSize = self.props.minimumSize
+
+				local dockWidgetPluginGuiInfo = DockWidgetPluginGuiInfo.new(
+					self.props.initDockState,
+					self.props.active,
+					self.props.overridePreviousState,
+					floatingSize.X, floatingSize.Y,
+					minimumSize.X, minimumSize.Y
+				)
+
+				self.pluginGui = plugin:CreateDockWidgetPluginGui(self.props.id, dockWidgetPluginGuiInfo)
+
+				if self.props.onInitialState then
+					self.props.onInitialState(self.pluginGui.Enabled)
+				end
+
+				self.pluginGui:BindToClose(function()
+					if self.props.onClose then
+						self.props.onClose()
+					else
+						self.pluginGui.Enabled = false
+					end
+				end)
+			else
+				-- Make sure the initial state is preserved until something changes
+				self.pluginGui.Enabled = self.props.active
+			end
```

Comment:
Can this happen in `init` instead of conditionally in `render`? We should be able to break this into two components, plus maybe a little function glue component to make this work out.

Reply:
Alright, will do.